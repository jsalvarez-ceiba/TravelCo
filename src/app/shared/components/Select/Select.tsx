import React, { SelectHTMLAttributes, FC } from 'react';

interface SelectProps extends SelectHTMLAttributes<HTMLSelectElement> {
  name: string;
  data: any[];
}

interface Places {
    city: string;
    country: string;
    flag: string;
}

const Select: FC<SelectProps> = ({ name, data, title, ...rest }: SelectProps) => {
  data = [];
  return (
    <select name={name} {...rest} className="form-control">
      <option value="">Seleccionar {title} </option>
        {
            data.map((element : Places , index : number) => (
                <option key={index.toString()} value={element.city} > {element.city} </option>
            ))
        }
    </select>
  );
};
export default Select;
