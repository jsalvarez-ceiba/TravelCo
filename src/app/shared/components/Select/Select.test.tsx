import React from 'react';
import renderer from 'react-test-renderer';
import Select from './Select';

describe('Test Select Component', () => {

    const select = renderer.create(<Select />).toJSON();

    test('Select Render', () => {
        expect(select).toMatchSnapshot();
    });

    test('Select Render', () => {
        expect(true).toBeTruthy();
    });

});