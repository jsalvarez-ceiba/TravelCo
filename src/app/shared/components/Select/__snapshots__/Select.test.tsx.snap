// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select Component Select Render 1`] = `
<select
  className="form-control"
>
  <option
    value=""
  >
    Seleccionar 
     
  </option>
</select>
`;
